### PR TITLE
Cancel Vote System

### PR DESCRIPTION
**Pull request by Ultradragon005**
The first vote is for skill and will be returned.
The second vote will be your real vote.

Role to work with

- [x]  Cleanser
- [x]  Eraser
- [x]  Fortune teller
- [x]  Tracker
- [x]  Keeper 
- [x]  Oracle
- [x]  Godfather
- [x] Soul Collector

### CODE REVIEW

**Comment by NikoCat233** [comment]
Dont merge yet, its buggy

**Comment by NikoCat233** [comment]
You should add a option setting to allow /vote commands
and make /vote 253 as skip vote (just do castvote 253)

also a setting to spam chats on received /vote

**Comment by NikoCat233** [comment]
Ok it seems good

**Comment by NikoCat233** [comment]
image
Can u do this? I think it makes more sense

**Comment by Tommy-XL** [comment]
I think merge this after dev 17 or until there is a more or less stable dev version

**Comment by Tommy-XL** [comment]
And also, some roles have a “hide vote” setting, I think now it doesn't make any sense

**Comment by Ultradragon005** [comment]
> And also, some roles have a “hide vote” setting, I think now it doesn't make any sense

oh right. gonna just remove that.

File: Patches/MeetingHudPatch.cs [hunk 1]
```diff
@@ -190,8 +190,6 @@ public static bool Prefix(MeetingHud __instance)
                 var player = Utils.GetPlayerById(ps.TargetPlayerId);
                 var playerRoleClass = player.GetRoleClass();
 
-                // Hide roles vote
-                if (playerRoleClass.HideVote(ps)) continue;
```
Comment (Tommy-XL) [inline Patches/MeetingHudPatch.cs:194]: I think need leave this feature as some new roles may use this

Or just comment out this code so as not to lose it

Comment (Ultradragon005) [reply Patches/MeetingHudPatch.cs:194]: hmm okay

**Comment by Tommy-XL** [comment]
I think this system can be improved, for example if the player does not want to use the ability he can click on skip voting and go to the real voting

**Comment by Tommy-XL** [comment]
Can you fix conflict?

**Comment by Ultradragon005** [comment]
fixed

**Comment by Ultradragon005** [comment]
image

**Pull request merged**